client/src/lib/session/context/TimerContext.ts

```diff
@@ -0,0 +1,10 @@
+import {createContext} from 'react';
+
+export type ProgressTimerContextProps = {
+  onLoad: (duration: number) => void;
+  onSeek: (currentTime: number) => void;
+};
+
+export const ProgressTimerContext = createContext<
+  ProgressTimerContextProps | undefined
+>(undefined);
```

Comment:
What is the reason to create a context instead of using zustand?

Reply:
I think cause you happened to mention that as a solution and I wanted to try with props and failed and remembered you mentioning context. I can pretend it's for the ephemeral nature of this state but that is not the real reason :)